### PR TITLE
11 Manage read only shares

### PR DESCRIPTION
**Pull request by FabianDevel**
Depends on #1419 

Disable all actions (multiselection, file, preview) if the public share doesn't allow downloading.

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/ui/fileList/preview/PreviewOtherFragment.kt [hunk 1]
```diff
@@ -46,6 +47,7 @@ class PreviewOtherFragment : PreviewFragment() {
 
         container.setOnClickListener { toggleFullscreen() }
 
+        bigOpenWithButton.isGone = file.isPublicShared() && !previewSliderViewModel.publicShareCanDownload
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/preview/PreviewOtherFragment.kt:50]: If `bigOpenWithButton` is gone, is it necessary to set the `onClickListener` ?

File: app/src/main/java/com/infomaniak/drive/ui/publicShare/PublicShareActivity.kt [hunk 2]
```diff
@@ -61,7 +63,7 @@ class PublicShareActivity : AppCompatActivity() {
             }
             else -> false
         }
-        binding.mainPublicShareButton.isVisible = isMainButtonVisible
+        binding.mainPublicShareButton.isVisible = isMainButtonVisible && publicShareViewModel.canDownloadFiles
```
Comment (tevincent) [inline app/src/main/java/com/infomaniak/drive/ui/publicShare/PublicShareActivity.kt:66]: Maybe we can merge this condition with `isMainButtonVisible` ?

**Review by tevincent — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/drive/ui/LaunchActivity.kt [hunk 3]
```diff
@@ -252,16 +245,17 @@ class LaunchActivity : AppCompatActivity() {
...
-        fileId: Int = -1,
+        shareLink: ShareLink? = null,
         isPasswordNeeded: Boolean = false,
         isExpired: Boolean = false,
     ) {
         publicShareActivityExtras = PublicShareActivityArgs(
             driveId = driveId.toInt(),
             publicShareUuid = publicShareUuid,
-            fileId = fileId,
+            fileId = shareLink?.fileId ?: -1,
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/ui/LaunchActivity.kt:255]: Add a sentry if needed

File: app/src/main/java/com/infomaniak/drive/ui/publicShare/PublicShareViewModel.kt [hunk 4]
```diff
@@ -61,18 +62,25 @@ class PublicShareViewModel(val savedStateHandle: SavedStateHandle) : ViewModel()
...
+    fun initPublicShare(onSuccess: (Int?, Boolean) -> Unit, onError: (ApiError?) -> Unit) {
+        viewModelScope.launch(Dispatchers.IO) {
+            val apiResponse = ApiRepository.getPublicShareInfo(driveId, publicShareUuid)
+            Dispatchers.Main {
+                when (apiResponse.result) {
+                    ApiResponseStatus.SUCCESS -> apiResponse.data?.let {
+                        onSuccess(it.fileId, it.capabilities?.canDownload ?: false)
+                    } ?: onError(null)
+                    else -> onError(apiResponse.error)
+                }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/ui/publicShare/PublicShareViewModel.kt:83]: Refactor with a result livedata

**Review by sirambd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/drive/ui/LaunchActivity.kt (hunk 3 above)
Comment (FabianDevel) [reply app/src/main/java/com/infomaniak/drive/ui/LaunchActivity.kt]: We don't need a Sentry, it's the attended behaviour.
If the link is protected by a password or not available anymore, we won't have a sharelink, and thus no fileId

**Review by tevincent — APPROVED** [review]

**Review by sirambd — APPROVED** [review]

**Pull request merged**